Add sample for custom routing with Kong Gateway (#4110)
knative/docs: docs/serving/samples/kong-routing-go/README.md

---
title: "Routing across multiple Knative services with Kong"
linkTitle: "Routing services with Kong"
weight: 1
type: "docs"
---

> 

Reviewer: We don't include this frontmatter anymore, so it can be removed

> # Routing across multiple Knative services with Kong

Reviewer: ```suggestion
# Routing across multiple Knative Services with Kong
```

> # Routing across multiple Knative services with Kong
> 
> This example shows how to map multiple Knative services to different paths under
> a single domain name using [Kong Gateway](https://docs.konghq.com/gateway-oss/).
> Kong Gateway is a general-purpose reverse proxy and API gateway built on
> [Nginx](https://www.nginx.com/), therefore these directions can also be used to
> configure routing based on other request data such as headers, or even to map
> Knative and external resources under the same domain name.
> 
> In this sample, we set up two web services: _Search_ service and _Login_
> service, which simply read in an env variable `SERVICE_NAME` and print
> `"${SERVICE_NAME} is called"`. We'll then define routing rules via [Kong's
> Kubernetes Ingress
> Controller](https://docs.konghq.com/kubernetes-ingress-controller/) and [Kong's
> request transformer
> plugin](https://docs.konghq.com/hub/kong-inc/request-transformer/), so that
> `example.com/search` maps to the _Search_ service, and `example.com/login` maps
> to the _Login_ service.

Reviewer: ```suggestion
# Routing across multiple Knative Services with Kong

This example shows how to map multiple Knative Services to different paths under
a single domain name using [Kong Gateway](https://docs.konghq.com/gateway-oss/).
Kong Gateway is a general-purpose reverse proxy and API gateway built on
[Nginx](https://www.nginx.com/), therefore these directions can also be used to
configure routing based on other request data such as headers, or even to map
Knative and external resources under the same domain name.

This sample sets up two web services: _Search_ service and _Login_
service, which read an environment variable called `SERVICE_NAME`, and print
`"${SERVICE_NAME} is called"`. The sample then defines routing rules by using [Kong's
Kubernetes Ingress
Controller](https://docs.konghq.com/kubernetes-ingress-controller/) and [Kong's
request transformer
plugin](https://docs.konghq.com/hub/kong-inc/request-transformer/), so that
`example.com/search` maps to the _Search_ service, and `example.com/login` maps
to the _Login_ service.
```
